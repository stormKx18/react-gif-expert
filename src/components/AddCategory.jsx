import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        setInputValue(target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        const cleanInput = inputValue.trim()
        if(cleanInput.length<=1) return;
        //setCategories((categories)=>[inputValue,...categories]);
        onNewCategory(cleanInput)
        setInputValue('')
    }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
        <input 
        type="text" 
        placeholder="Search gifs..."
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}

