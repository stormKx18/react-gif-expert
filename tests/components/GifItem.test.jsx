import { fireEvent, render, screen  } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Tests for <GifItem />', () => {

    const title= 'Test Title'
    const url  = 'https://www.awewomewebpage.comp/'

    test('Component should match snapshot ', () => {
      const {container} = render(<GifItem title={title} url={url}/>)
      expect(container).toMatchSnapshot();
    })
    
    test('Component should show the title ', () => {
      render(<GifItem title={title} url={url}/>)
      expect(screen.getByText(title)).toBeTruthy();
    })

    test('Title should not react to click ', () => {
        render(<GifItem title={title} url={url}/>)
        fireEvent.click(screen.getByText(title))
        //screen.debug()
        expect(screen.getByText(`${title}`)).toBeTruthy();
    })

    test('Component should show image with URL and ALT text', () => {
        render(<GifItem title={title} url={url}/>)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url);    
        // expect(screen.getByRole('img').alt).toContain(title); 
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toContain(title);   
           
    })


    
    
  
})

