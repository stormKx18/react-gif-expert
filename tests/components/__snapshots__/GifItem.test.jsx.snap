// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifItem /> Component should match snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="https://www.awewomewebpage.comp/"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
